scripts/isolate_tests.py

```diff
@@ -106,13 +84,18 @@ def extract_and_write(f, path):
     write_cases(f, cases)
 
 if __name__ == '__main__':
+    if len(sys.argv) == 1:
+        print("Usage: " + sys.argv[0] + " path-to-file-or-folder-to-extract-code-from [docs]")
+        exit(1)
+
     path = sys.argv[1]
     docs = False
     if len(sys.argv) > 2 and sys.argv[2] == 'docs':
         docs = True
 
     if isfile(path):
-        extract_and_write(path, path)
+        _, tail = split(path)
+        extract_and_write(tail, path)
```

Comment:
Doesn't this break the script? I think that if I run `./isolate_tests.sh docs/types/conversion.rst` this change will make it look for `conversion.rst` in the current directory rather than for `docs/types/conversion.rst` as it should.

Reply:
No, because it still uses `path` to find the file and only uses `tail` to write out the file which was what was failing before this PR

Reply:
Ah, right.

I wanted to say that in that case there can probably be name collisions if two files with the same name are in multiple directories and we need to check that we're not ovewriting files. But I see that even then the name is only used as input for the hash together with file content. It's fine then.

Reply:
No, `path` is always the full path (relative or absolute), the first parameter (here `tail`) is just for writing the result.
I'll change this to one parameter in my next push as it seems needlessly complicated this way anyway.